src/kernel/tty.zig

```diff
@@ -1,437 +1,42 @@
-            row += 1;
-            scroll();
-        },
-        '\t' => {
-            column += 4;
-            if (column >= vga.WIDTH) {
-                column -= @truncate(u8, vga.WIDTH);
-                row += 1;
-                scroll();
-            }
-        },
-        '\r' => {
-            column = 0;
-        },
-        // \b
-        '\x08' => {
-            if (column == 0) {
-                if (row != 0) {
-                    column = vga.WIDTH - 1;
-                    row -= 1;
-                }
-            } else {
-                column -= 1;
-            }
-        },
-        else => {
-            try putEntryAt(char, column, row);
-            column += 1;
-            if (column == vga.WIDTH) {
-                column = 0;
-                row += 1;
-                scroll();
-            }
-        },
-    }
-}
-
-///
-/// Print a string to the TTY. This also updates to hardware cursor.
-///
-/// Arguments:
-///     IN str: []const u8 - The string to print.
-///
-/// Errors:
-///     TtyError.OutOfBounds - If trying to print beyond the video buffer.
-///
-fn writeString(str: []const u8) TtyError!void {
-    // Make sure we update the cursor to the last character
-    defer updateCursor();
-    for (str) |char| {
-        try putChar(char);
-    }
-}
-
-///
-/// Print the pluto logo.
-///
-fn printLogo() void {
-    const column_temp = column;
-    const row_temp = row;
-
-    defer column = column_temp;
-    defer row = row_temp;
-
-    const logo =
-        \\                  _____    _        _    _   _______    ____
-        \\                 |  __ \  | |      | |  | | |__   __|  / __ \
-        \\                 | |__) | | |      | |  | |    | |    | |  | |
-        \\                 |  ___/  | |      | |  | |    | |    | |  | |
-        \\                 | |      | |____  | |__| |    | |    | |__| |
-        \\                 |_|      |______|  \____/     |_|     \____/
-    ;
-
-    // Print the logo at the top of the screen
-    column = 0;
-    row = 0;
-
-    writeString(logo) catch |e| {
-        log.logError("TTY: Error print logo. Error {}\n", .{e});
-    };
-}
+/// The current tty stream
+var tty: TTY = undefined;
 
 ///
-/// A call back function for use in the formation of a string. This calls writeString normally.
+/// A call back function for use in the formation of a string. This calls the architecture's print function.
 ///
 /// Arguments:
 ///     IN ctx: void       - The context of the printing. This will be empty.
 ///     IN str: []const u8 - The string to print.
 ///
+/// Return: usize
+///     The number of characters printed
+///
 /// Errors:
 ///     TtyError.OutOfBounds - If trying to print beyond the video buffer.
 ///
-fn printCallback(ctx: void, str: []const u8) TtyError!usize {
-    try writeString(str);
+fn printCallback(ctx: void, str: []const u8) !usize {
+    tty.print(str) catch |e| panic(@errorReturnTrace(), "Failed to print to tty: {}\n", .{e});
```

Comment:
My thinking is that it's not the caller's fault that something went wrong, it's something wrong with the TTY system rather than what the caller is trying to do, so therefore the caller shouldn't be given the responsibility to handle it.

Reply:
yes, good point, so can remove the error doc comment